### PR TITLE
Add additional provider details to ecr viewer

### LINKED ISSUE #2508: Update "Provider Details" in Encounter Info
## User Story

As a case investigator or eCR coordinator, I want to be able to see the specific provider information for the eCR encounter. This will help inform my investigation and the specific provider (doctor) at a facility I should follow up with.

## Acceptance Criteria

- [x] `Provider Details` section includes the data below in this order:
    - [x] Provider Name
    - [x] Provider Address
    - [x] Provider Contact
    - [x] Provider Facility Name
    - [x] Provider Facility Address
    - [x] Provider ID
- [x] The data in the fields should match the data found in the `componentOf > encompassingEncounter > responsibleParty > assignedEntity` section of the xml
- [x] If these fields are not present in the eCR document, they are added to the `Unavailable Info` section

## Scenarios

**Scenario 1:** 
- When viewing an eCR, I want to know the specific provider information that a patient saw during their encounter. 
- When I scroll down to the `Encounter Info` section, I can see `Provider Details` section in the Viewer
- The fields are listed in the order below:
  - Provider Name
  - Provider Address
  - Provider Contact
  - Provider Facility Name
  - Provider Facility Address
  - Provider ID
- The data that I can see in this section corresponds exactly to the data I see in the `componentOf > encompassingEncounter > responsibleParty > assignedEntity` section of the xml
- When I scroll down to `Unavailable Info` section, I see any fields we didn't have data for in the correct order

## Design Notes

[Link to Figma design in Dev mode](https://www.figma.com/design/zAPyOEYwNtx3f2Y47aoq0G/eCR-Viewer-(Aug-Dec-2024)?node-id=7006-93&node-type=CANVAS&m=dev)

_Screenshots below should not be used for working on this ticket - always refer to Figma design for most up-to-date designs:_

**this screenshot doesn't have ALL fields, refer to list above or spreadsheet for all fields**

## Additional context

There's a good deal of similar, author-related information within an eCR - facility, provider, author, encounter location, etc. We previously were displaying some Provider information, but not all.

This spreadsheet has more information about where in the eCR spec Provider information can be found: [eCR Viewer Field Mapping](https://cdc.sharepoint.com/:x:/r/teams/OPHDST-IRD-DIBBs/Shared%20Documents/CDC%20DIBBs%20Project/Streamline%20Team/1%20-%20Pilot%20eCR%20Viewer%202024/5%20-%20Research%20%26%20Design/Design/eCR%20Viewer%20Field%20Mapping.xlsx?d=w80a53c04215840b6a7762dfc8b7267c9&csf=1&web=1&e=RIiBPb)

area to look at in spreadsheet highlighted in red box:

### PR DESCRIPTION
**Pull request by BobanL**
# PULL REQUEST

## Summary
- Update fhir conversion to set encounter participant to be `PractitionerRole`, which includes a `Practitioner` and `Organization`.
  - Update `Practitioner` to remove any organization information
- Add additional provider fields to be displayed
- Add .5rem padding right to data title fields to support longer titles.

## Related Issue
Fixes #2508 

## Acceptance Criteria
- [x] `Provider Details` section includes the data below in this order:
    - [x] Provider Name
    - [x] Provider Address
    - [x] Provider Contact
    - [x] Provider Facility Name
    - [x] Provider Facility Address
    - [x] Provider ID
- [x] The data in the fields should match the data found in the `componentOf > encompassingEncounter > responsibleParty > assignedEntity` section of the xml
- [x] If these fields are not present in the eCR document, they are added to the `Unavailable Info` section

## Additional Information
image

## Checklist

- [ ] If this code affects the other scrum team, have they been notified? (In Slack, as reviewers, etc.)

[//]: # (PR title: Remember to name your PR descriptively!)

### CODE REVIEW

**Comment by BobanL** [comment]
@sarahtress - The field `Provider Facility Address` is a bit long. Is this fine or should something change?

image

**Comment by sarahtress** [comment]
> @sarahtress - The field `Provider Facility Address` is a bit long. Is this fine or should something change?
> 
> image

hmmm actually this feels like more of a "we need to add some padding or like a max width so that "Address" would go to the next line" problem

**Comment by BobanL** [comment]
If I add .5 rem padding on the right this is what it looks like. Does this
look correct?
[image: image.png]

**Comment by sarahtress** [comment]
@BobanL that looks good to me, thank you!

File: containers/fhir-converter/Templates/eCR/Header.liquid [hunk 1]
```diff
@@ -29,7 +29,19 @@
...
   {% include 'Resource/PractitionerResponsibleParty' practitioner: msg.ClinicalDocument.componentOf.encompassingEncounter.responsibleParty.assignedEntity ID: practitionerId -%}
   {% assign fullPractitionerId = practitionerId | prepend: 'Practitioner/' -%}
-  {% include 'Reference/Encounter/Participant_ResponsibleProvider' ID: encounterId, REF: fullPractitionerId -%}
+  {% assign practitionerRoleId =  msg.ClinicalDocument.componentOf.encompassingEncounter.responsibleParty.assignedEntity | to_json_string | generate_uuid %}
+  {% assign fullPractitionerRoleId = practitionerRoleId | prepend: 'PractitionerRole/' %}
+  {% include 'Reference/PractitionerRole/Practitioner', REF: fullPractitionerId, ID: practitionerRoleId -%}
+
+  {% include 'Reference/Encounter/Participant_ResponsibleProvider' ID: encounterId, REF: fullPractitionerRoleId -%}
+  {% if msg.ClinicalDocument.componentOf.encompassingEncounter.responsibleParty.assignedEntity.representedOrganization %}
+    {% evaluate organizationId using 'Utils/GenerateId' obj: msg.ClinicalDocument.componentOf.encompassingEncounter.responsibleParty.assignedEntity.representedOrganization -%}
```
Comment (angelathe) [inline containers/fhir-converter/Templates/eCR/Header.liquid:38]: Curious, what's the difference between using `Utils/GenerateId` and using the `generate_uuid` custom filter?

**Review by angelathe — APPROVED** [review]
LGTM! 🎉 
(Compared provider details of several eCRs against their eCR XML files)

File: containers/fhir-converter/Templates/eCR/Header.liquid (hunk 1 above)
Comment (BobanL) [reply containers/fhir-converter/Templates/eCR/Header.liquid:38]: So the [`evaluate`](https://github.com/microsoft/FHIR-Converter/blob/c7bdbaf922499310da224e1db855b936f8919df0/docs/Filters-and-Tags.md#:~:text=Syntax-,evaluate,-Evaluates%20an%20ID) is custom tag that will run another liquid template on a value and then assign it. The [`Utils/GenerateId`](https://github.com/microsoft/FHIR-Converter/blob/c7bdbaf922499310da224e1db855b936f8919df0/data/Templates/Ccda/Utils/_GenerateId.liquid#L4) is a liquid file that calls the [generate_uuid filter](https://github.com/microsoft/FHIR-Converter/blob/c7bdbaf922499310da224e1db855b936f8919df0/src/Microsoft.Health.Fhir.Liquid.Converter/Filters/GeneralFilters.cs#L76) with additional conditionals around which id it selects from the object.

Here's a [link to the code for the evaluate custom tag](https://github.com/microsoft/FHIR-Converter/blob/c7bdbaf922499310da224e1db855b936f8919df0/src/Microsoft.Health.Fhir.Liquid.Converter/DotLiquids/Evaluate.cs#L17), but I don't totally don't understand it

**Pull request merged**